lib/ast.js

```diff
@@ -308,6 +308,19 @@ var AST_Scope = DEFNODE("Scope", "variables functions uses_with uses_eval parent
         enclosed: "[SymbolDef*/S] a list of all symbol definitions that are accessed from this scope or any subscopes",
         cname: "[integer/S] current index for mangling variables (used internally by the mangler)",
     },
+    clone: function(deep) {
+        var node = this._clone(deep);
+        node.variables = new Dictionary();
+        this.variables.each(function(value, key) {
+            node.variables.set(key, value);
+        });
+        node.functions = new Dictionary();
+        this.functions.each(function(value, key) {
+            node.functions.set(key, value);
+        });
```

Comment:
Should create a `Dictionary.prototype.clone` method.

Reply:
Sorry for duplicates. Github GUI issue.

Reply:
Done in https://github.com/mishoo/UglifyJS2/pull/2803/commits/3b679cd5d2427a0382815ad40459d6f95109aac0

Reply:
> Sorry for duplicates. Github GUI issue.

I almost thought you also want a `Comment.prototype.clone` method :ghost: